client/starwhale/cli/assistance/broker.py

```diff
@@ -0,0 +1,288 @@
+    def __init__(self) -> None:
+        self.sessions: Dict[str, Session] = {}
+        self.garbage_collector: Optional[PeriodicRunner] = None
+        self.keep_alive_monitor: Optional[PeriodicRunner] = None
+
+    def keep_alive_check(self) -> None:
+        with global_lock:
+            now = datetime.now().timestamp()
+            for session in self.sessions.values():
+                for command in session.commands.values():
+                    if now - command.host_access_time > settings.keep_alive_seconds:
+                        command.get_file("stdin").close("host timeout")
+                        command.get_file("stdout").close("host timeout")
+                        command.get_file("stderr").close("host timeout")
+                    if now - command.remote_access_time > settings.keep_alive_seconds:
+                        command.get_file("stdin").close("remote timeout")
+                        command.get_file("stdout").close("remote timeout")
+                        command.get_file("stderr").close("remote timeout")
+
+    def garbage_collect(self) -> None:
+        with global_lock:
+            now = datetime.now().timestamp()
+            new_sessions = {}
+            for session_id, session in self.sessions.items():
+                if now - session.last_access_time > settings.gc_timeout_seconds:
+                    print(f"garbage collection: remove session {session_id}")
+                else:
+                    new_sessions[session_id] = session
+                    new_commands = {}
+                    for command_id, command in session.commands.items():
+                        if now - command.last_access_time > settings.gc_timeout_seconds:
+                            print(f"garbage collection: remove command {command_id}")
+                        else:
+                            new_commands[command_id] = command
+                    session.commands = new_commands
+            self.sessions = new_sessions
+
+    def start_garbage_collector(self) -> None:
+        with global_lock:
+            if self.garbage_collector is None:
+                self.garbage_collector = PeriodicRunner(
+                    self.garbage_collect, settings.gc_interval_seconds
+                )
+                self.garbage_collector.start()
+                print("garbage collector started")
+
+    def stop_garbage_collector(self) -> None:
+        with global_lock:
+            if self.garbage_collector is not None:
+                self.garbage_collector.stop()
+                self.garbage_collector = None
+
+    def start_keep_alive_monitor(self) -> None:
+        with global_lock:
+            if self.keep_alive_monitor is None:
+                self.keep_alive_monitor = PeriodicRunner(
+                    self.keep_alive_check, settings.keep_alive_check_interval_seconds
+                )
+                self.keep_alive_monitor.start()
+                print("keep alive monitor started")
```

Comment:
replace `print` with `loguru.logger.info` ?

Reply:
For broker, I guess print is enough. It is a very simple service.